libmemcached/response.cc

```diff
@@ -749,6 +749,35 @@ static bool parse_response_header(char *buffer,
   return true;
 }
 
+static bool parse_response_size_value(char *buffer, int buffer_length,
+                                      int size_count, size_t *size_array)
+{
+  char *string_ptr= buffer;
+  char *end_ptr= buffer + buffer_length;
+
+  for (int i=0; i<size_count; i++)
+  {
+    if (*string_ptr != ' ' && *string_ptr != '=')
+        return false;
+    string_ptr++;
+
+    char *start_ptr= string_ptr;
+    for ( ; string_ptr != end_ptr && isdigit(*string_ptr); string_ptr++) {};
+
+    if (string_ptr ==  start_ptr || string_ptr >= end_ptr)
+      return false;
+
+    size_array[i]= (size_t)strtoul(start_ptr, &string_ptr, 10);
+  }
+
+  if (*string_ptr != '\r' || *(string_ptr+1) != '\n')
+  {
+    return false;
+  }
```

Comment:
/* Skip spaces */
  if (*str_ptr == '\r' && (str_ptr+1) < end_ptr && *(str_ptr+1) == '\n')
  {
    str_ptr+= 2; /* Skip past the \r\n */
  }